var/spack/repos/builtin/packages/bzip2/package.py

```diff
@@ -46,6 +46,14 @@ def patch(self):
             filter_file('-Wall -Winline', '-Minform=inform', 'Makefile')
             filter_file('-Wall -Winline', '-Minform=inform', 'Makefile-libbz2_so')  # noqa
 
+        # The Makefiles use GCC flags that are incompatible with XL and nvcc
+        if self.compiler.name == 'xl':
+            filter_file('-fpic', '', 'Makefile')
+            filter_file('-fpic', '', 'Makefile-libbz2_so')
+            filter_file('bzip2.c libbz2.so.1.0.6',
```

Comment:
this version looks so hard-coded :)

Reply:
Yes, the included `Makefile-libbz2_so` is really hard-coded :) In order to filter and catch only that part of the one line I had to write it that way. It would really be best to rewrite the Makefile entirely, but that's something that needs to be done upstream.

Reply:
Okay, but can we replace the `1.0.6` with something like `self.spec.version`? (not entirely sure if that's the right attribute)

Reply:
Done! Thanks